### PR TITLE
Request and response logging

### PR DESCRIPTION
**Pull request by christophermanahan**
### Description
- Implement logging middleware for all requests received by and responses sent by the HTTP Server

### CODE REVIEW

File: src/main/java/io/github/christophermanahan/carnitas/HTTPServer.java [hunk 1]
```diff
@@ -29,7 +29,9 @@ private Runnable connect(Listener listener) {
 
     private void serve(Connection connection) {
         parser.parse(new ConnectionReader(connection))
+          .map(logger::log)
           .map(handler::handle)
+          .map(logger::log)
```
Comment (damonkelley) [inline src/main/java/io/github/christophermanahan/carnitas/HTTPServer.java]: Let's think about how we could simplify this.

Hint: Think about logging as something that the developer could configure (i.e. making it part of the application).

Hint: Middleware 😄

File: src/main/java/io/github/christophermanahan/carnitas/Response.java [hunk 2]
```diff
@@ -48,4 +52,17 @@ public boolean equals(Object object) {
...
           && body.equals(response.body);
     }
+
+    @Override
+    public String toString() {
+        return Stream.of(statusLine(), headers, List.of(body.orElse("")))
+          .flatMap(Collection::stream)
+          .filter(s -> !s.isEmpty())
+          .collect(Collectors.joining(Serializer.CRLF))
+          .concat(Serializer.CRLF);
```
Comment (damonkelley) [inline src/main/java/io/github/christophermanahan/carnitas/Response.java]: This introduces a circular dependency between the Response and the Serializer, yeah?

File: src/main/java/io/github/christophermanahan/carnitas/LoggingMiddleware.java [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    private final MessageLogger logger;
+    private final Handler handler;
+
+    LoggingMiddleware(Handler handler) {
+        this.logger = new MessageLogger();
+        this.handler = handler;
+    }
+
+    public Response handle(Request request) {
+        logger.log(request.toString());
```
Comment (damonkelley) [inline src/main/java/io/github/christophermanahan/carnitas/LoggingMiddleware.java]: Usually `toString` is used for debugging purposes, not for presentation logic.

Also, we don't need the whole request in the log. For the request, it is fine to just have something like

```
METHOD /path HTTP/1.1
```
and for response

```
200 OK
```

File: src/main/java/io/github/christophermanahan/carnitas/Response.java (hunk 2 above)
Comment (christophermanahan) [reply src/main/java/io/github/christophermanahan/carnitas/Response.java]: Response now contains the CRLF constant.

**Pull request merged**